src/AutoMapper/MapperConfiguration.cs

```diff
@@ -66,25 +66,66 @@ public MapperConfiguration(Action<IMapperConfigurationExpression> configure, IEn
         {
             var key = new TypePair(typeof(TSource), typeof(TDestination));
             var mapRequest = new MapRequest(key, types);
-            return (Func<TSource, TDestination, ResolutionContext, TDestination>)GetMapperFunc(mapRequest);
+            var mapperFuncs = GetMapperFuncs(mapRequest);
+            return GetSubMap<TSource, TDestination>(mapperFuncs)?.Compile() ?? (Func<TSource, TDestination, ResolutionContext, TDestination>)mapperFuncs.Typed;

```

Comment:
I think this sub map this should be inside the MapperFunc. This way is not even clear when the compile happens.